ivy/functional/backends/mxnet/general.py

```diff
@@ -246,11 +246,11 @@ def one_hot(indices, depth, device=None):
     return mx.nd.one_hot(indices, depth)
 
 
-def shape(x: mx.nd.NDArray, as_array: bool = False) -> Union[tuple, mx.nd.NDArray]:
+def shape(x: mx.nd.NDArray, as_array: bool = False) -> Union[ivy.Shape, ivy.Array]:
     if as_array:
-        return mx.nd.shape_array(x)
+        return ivy.array(x.shape)
```

Comment:
actually we should have `ivy.array(mx.nd.shape_array(x))` or `ivy.Array(mx.nd.shape_array(x))` if `mx.nd.shape_array` returns a native shape instance.